src/tracker/mime-type-tracker-query.vala

```diff
@@ -34,13 +38,40 @@ public class Games.MimeTypeTrackerQuery : Object, TrackerQuery {
 		return false;
 	}
 
-	public Game game_for_cursor (Tracker.Sparql.Cursor cursor) throws Error {
+	public void process_cursor (Tracker.Sparql.Cursor cursor) {
 		var uri = cursor.get_string (0);
+		uris += uri;
+	}
+
+	public async void foreach_game (GameCallback game_callback) {
+		uint handled_uris = 0;
+		foreach (var uri in uris) {
+			try {
+				var game = game_for_uri (uri);
+				game_callback (game);
+				handled_uris++;
+
+				// Free the execution only once every HANDLED_URIS_PER_CYCLE
+				// games to speed up the execution by avoiding too many context
+				// switching.
+				if (handled_uris >= HANDLED_URIS_PER_CYCLE) {
+					handled_uris = 0;
+
+					Idle.add (this.foreach_game.callback);
+					yield;
+				}
+			}
+			catch (Error e) {
+				debug (e.message);
+			}
+		}
+	}
 
+	public Game game_for_uri (string uri) throws Error {
```

Comment:
This method before you edit it:
- retrieved the URI from the cursor,
- checked that the URI was valid and thrown an error if it wasn't,
- uried the URI.

Now we process the cursor earlier, we don't want to throw errors but instead just do nothing.

Having a private method simply wrapping a call to a private callback of the same name that we are guaranteed to use only is one place looks like bloat to me: the fact that the method and the callback have the same name and are both private is a good indicator that it is bloat IMO.